### DISCUSSION TITLE
🐛 Bug Report: Unable to connect to Remote Database

### DISCUSSION THREAD

**Discussion by kennethmakjw** [appwrite/appwrite#4644] [Q&A]
### 👟 Reproduction steps

I am trying to connect my remote DB from DigitalOcean Managed Database and I'm not able to connect to it because Appwrite failed to established a connection.

### 👍 Expected behavior

It should connect to the database and Appwrite should operate normally.

### 👎 Actual Behavior

Appwrite did not work, it failed to connect, the logs shows:
```
Fatal error: Uncaught Exception: Failed to connect to database: SQLSTATE[HY000] [1105] unknown error: Code: UNAVAILABLE
server does not allow insecure connections, client must use SSL/TLS
 in /usr/src/code/app/http.php:77
Stack trace:
#0 {main}
  thrown in /usr/src/code/app/http.php on line 77
[2022-11-06 18:42:13 #1.2]      ERROR   php_swoole_server_rshutdown() (ERRNO 503): Fatal error: Uncaught Exception: Failed to connect to database: SQLSTATE[HY000] [1105] unknown error: Code: UNAVAILABLE
server does not allow insecure connections, client must use SSL/TLS
 in /usr/src/code/app/http.php:77
Stack trace:
#0 {main}
  thrown in /usr/src/code/app/http.php on line 77
Database not ready. Retrying connection (10)...
```

### 🎲 Appwrite version

Version 1.0.x

### 💻 Operating system

Linux

### 🧱 Your Environment

I am trying to use a remote database, I have modified the .env and ran docker compose up -d

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@kennethmakjw is digital ocean's managed database MySQL? If so, that may be the problem. Appwrite uses mariadb and since Mariadb and MySQL are slightly different it won't work. Would you be able to use a mariadb database? Otherwise, you may need to tweak some of the Appwrite code (we can advise on that if needed).

**Reply by stnguyen90** [reply]
In short, you need to modify the code before Appwrite starts in all containers. There are ~7 places in the code with this line:

https://github.com/appwrite/appwrite/blob/44ca7d9459dba9b21760e2934decbfd41f26c1ba/app/init.php#L55

That needs to be replaced with:

```php
use Utopia\Database\Adapter\MySQL as MariaDB; 
```

I think that should be it, but I didn't get a chance to fully test it.

**Comment by kennethmakjw** [comment]
> @kennethmakjw is digital ocean's managed database MySQL? If so, that may be the problem. Appwrite uses mariadb and since Mariadb and MySQL are slightly different it won't work. Would you be able to use a mariadb database? Otherwise, you may need to tweak some of the Appwrite code (we can advise on that if needed).

Yes, it is indeed a MySQL server. 

Could you share the steps to tweak the code? 

Thanks in advance.
